models/property.py

```diff
@@ -1,10 +1,9 @@
 from db import db
 from utils.time import Time
-from marshmallow import ValidationError
 
 from models.tenant import TenantModel
 from models.base_model import BaseModel
-from models.user import UserModel, RoleEnum
+from models.user import UserModel
 from models.property_assignment import PropertyAssignment
```

Comment:
PropertyAssignment is used on line 24 when defining the relationship. I'm not sure, but maybe we could use the class name on line 24 instead of the table name; otherwise you'll just have to ignore what your editor is telling you.

Reply:
I tried putting in the class name, but no go. I remember looking in the docs to see if we had to use a the table name string representation, when reviewing the PR for that, however I was unsuccessful in finding another way to define the many-to-many relationship.

Reply:
Oh interesting, I didn't catch that.  That seems really strange...I feel like there _should_ be a better way to do it, but there may not be.  I'll make a note for myself to look into it, but for now I guess that's how it has to be 🤷‍♂️

Reply:
Passing in `PropertyAssignment.__tablename__` instead of the string works.  It's kind of ugly though.  Maybe would could assign it to a variable or something to make it look nicer and go that route?

Reply:
@NickSchimek, I was thinking about this 👆 over the holiday.

What do you think about adding the following code to the `BaseModel`?

```python
@classmethod
def get_tablename(cls):
    return cls.__tablename__
```

I would prefer this over over importing the model and then only using the string representation of the table name.  I think it's less confusing to see `secondary=PropertyAssignment.get_tablename()` or something similar because it exposes what's taking place and Flake8 won't flag the unused import.

Reply:
Very nice! This is a much better solution. I don't think I'd have any issue doing `PropertyAssignment.__tablename__` although we can also put it in the BaseModel as well, although I'd prefer to do `tablename` as the method name. `PropertyAssignment.tablename`.